core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -661,11 +661,21 @@ class KafkaApis(val requestChannel: RequestChannel,
     val versionId = request.header.apiVersion
     val clientId = request.header.clientId
     val fetchRequest = request.body[FetchRequest]
+    val (topicIds, topicNames) =
+      if (fetchRequest.version() >= 13)
+        metadataCache.topicIdInfo()
+      else
+        (Collections.emptyMap[String, Uuid](), Collections.emptyMap[Uuid, String]())
+
     val fetchContext = fetchManager.newContext(
```

Comment:
not sure whether this idea is valid. Could we resolve all ids to names and then keep using previous code to handle fetch requests?  for example: the code of deleting topic resolve ids to names first and then process the deletion by names. That is a good pattern to me as we don't need to trace id/name everywhere.

Reply:
For the most part, this is what we do. But we also have to keep track of unresolved names right?  FetchSession makes this more complicated. We may get a topic ID that we can't resolve, but with a subsequent update we can. I'm not 100% sure of the logistics of fetch sessions, but it doesn't seem like a good practice to leave unresolved IDs out of the session.

Reply:
I ended up deciding to end the session and throw a top level error when we have an unknown topic ID.